compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -6897,11 +6896,16 @@ private BType checkInvocationParam(BLangInvocation iExpr, AnalyzerData data) {
                             incRecordAllowAdditionalFields = false;
                         }
                         iExpr.requiredArgs.add(expr);
+                    } else if (invokableSymbol.restParam != null &&
+                            invokableSymbol.restParam.name.value.equals(((BLangNamedArgsExpression) expr).name.value)) {
+                        // can not provide a rest parameters as named args
+                        checkTypeParamExpr(expr, symTable.noType, iExpr.langLibInvocation, data);
+                        dlog.error(expr.pos, DiagnosticErrorCode.REST_ARG_CANNOT_BE_NAMED_ARG, iExpr.name.value);
                     } else {
                         if (incRecordAllowAdditionalFields && !namedArgForIncRecordParam) {
                             iExpr.requiredArgs.add(expr);
                         } else {
-                            checkTypeParamExpr(expr, new BNoType(TypeTags.NONE), iExpr.langLibInvocation, data);
+                            checkTypeParamExpr(expr, symTable.noType, iExpr.langLibInvocation, data);
                             dlog.error(expr.pos, DiagnosticErrorCode.TOO_MANY_ARGS_FUNC_CALL, iExpr.name.value);
                         }
```

Comment:
I would suggest to refactor like this.
```java
                    } else if (incRecordAllowAdditionalFields && !namedArgForIncRecordParam) {
                        iExpr.requiredArgs.add(expr);
                    } else {
                        boolean referringToARest = invokableSymbol.restParam != null && invokableSymbol.restParam.name
                                .value.equals(((BLangNamedArgsExpression) expr).name.value);
                        DiagnosticErrorCode errorCode = referringToARest ?
                                DiagnosticErrorCode.REST_ARG_CANNOT_BE_NAMED_ARG
                                : DiagnosticErrorCode.TOO_MANY_ARGS_FUNC_CALL;
                        checkTypeParamExpr(expr, symTable.noType, iExpr.langLibInvocation, data);
                        dlog.error(expr.pos, errorCode, iExpr.name.value);
                    }
```

Reply:
Wouldn't `else if` also need to have `!referringToARest` check to preserve the logic?

Reply:
!referringToARest case, we currently execute,
```java
else {
         checkTypeParamExpr(expr, symTable.noType, iExpr.langLibInvocation, data);
         dlog.error(expr.pos, DiagnosticErrorCode.TOO_MANY_ARGS_FUNC_CALL, iExpr.name.value);
}
```

With above aren't we doing the same?

Reply:
since `else` executes only if `(invokableSymbol.restParam != null && invokableSymbol.restParam.name.value.equals(((BLangNamedArgsExpression) expr).name.value))` (i.e. `referringToARest`) is false and 
```
if (incRecordAllowAdditionalFields && !namedArgForIncRecordParam) {
    iExpr.requiredArgs.add(expr);
}
```
exists inside the else block we can't have that alone right?

Reply:
```
} else if (incRecordAllowAdditionalFields && !namedArgForIncRecordParam && !referringToARest) {
    iExpr.requiredArgs.add(expr);
} else {
    DiagnosticErrorCode errorCode = referringToARest ?
        DiagnosticErrorCode.REST_ARG_CANNOT_BE_NAMED_ARG
            : DiagnosticErrorCode.TOO_MANY_ARGS_FUNC_CALL;
    checkTypeParamExpr(expr, symTable.noType, iExpr.langLibInvocation, data);
    dlog.error(expr.pos, errorCode, iExpr.name.value);
}
```
With 
```
boolean referringToARest = invokableSymbol.restParam != null && invokableSymbol.restParam.name
                                .value.equals(((BLangNamedArgsExpression) expr).name.value);
```

Reply:
Here in the `incRecordAllowAdditionalFields && !namedArgForIncRecordParam && !referringToARest`, I think last check you suggest is unnecessary, bcz when we allow additional fields in the included record, we can't have a rest. (i.e. rest values goes to the included record as per the above example)